tests/epics/areadetector/test_hdf_streamer_det.py

```diff
@@ -103,9 +51,10 @@ async def hdf_streamer_dets():
     yield deta, detb
 
 
-async def test_hdf_streamer_dets_step(hdf_streamer_dets: List[HDFStreamerDet], RE):
-    d = DocHolder()
-    RE(bp.count(hdf_streamer_dets), d.append)
+async def test_hdf_streamer_dets_step(
+    hdf_streamer_dets: List[HDFStreamerDet], RE, doc_holder
```

Comment:
Not a big fan of pytest fixtures if they don't need teardown support, as they complicate typing. I guess you could:
```suggestion
    hdf_streamer_dets: List[HDFStreamerDet], RE, doc_holder: DocHolder
```
But then you would need:
```python
from .conftest import DocHolder
``
so you might as well just create a `DocHolder` manually like the original code does...

Reply:
I just liked the idea of splitting up the tests. Currently, two test files need the DocHolder which is why I put them as a fixture. You are right; it does complicate the typing... perhaps just for the tests I'll do a bit of code duplication.

Reply:
Or stick it in a `test_utils.py` file and import it in the 2 test files?

Reply:
For the moment I just did what you initially suggested, and we have two DocHolders. If this becomes more widely used we can move it to a utility file in the tests.